File: mifosng-android/src/main/java/com/mifos/services/data/CollectionSheetPayload.java
package com.mifos.services.data;


import com.google.gson.Gson;


public class CollectionSheetPayload extends Payload
{
    public String actualDisbursementDate;
    public int[] bulkDisbursementTransactions;
    public BulkRepaymentTransactions[] bulkRepaymentTransactions;
    public String[] clientsAttendance;
     @Override
    public String toString()
    {
        return new Gson().toJson(this);
    }

}

File: core/network/src/main/java/com/mifos/core/network/model/CollectionSheetPayload.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.network.model

import com.mifos.room.entities.noncore.BulkRepaymentTransactions
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
class CollectionSheetPayload : Payload() {
    var actualDisbursementDate: String? = null
    var bulkDisbursementTransactions: List<Int> = emptyList()
    var bulkRepaymentTransactions: List<BulkRepaymentTransactions> = emptyList()
    var clientsAttendance: List<String> = emptyList()

    override fun toString(): String {
        return Json.encodeToString(this)
    }
}
